cpp/src/arrow/filesystem/azurefs.cc

```diff
@@ -119,6 +119,14 @@ Status AzureOptions::ConfigureDefaultCredential(const std::string& account_name)
   return Status::OK();
 }
 
+Status AzureOptions::ConfigureManagedIdentityCredential(const std::string& account_name,
```

Comment:
It seems that `account_name` isn't used. Is it needed?

Reply:
Good catch. Sorry, I have not been paying enough attention :man_facepalming: . I think @felipecrv refactored this. I need to check how the account name is configured now. I'll do it tomorrow.

Reply:
Thanks.
It seems that both of `ConfigureWorkloadIdentityCredential()` and `ConfigureDefaultCredential()` are the same situation. Could you also check them too?

Reply:
Sorry. I messed up and forgot to set `account_name_` on the `Configure...` functions.

Reply:
I think we should undo something I did and move `account_name_` back to a public `account_name` field (the first in `AzureOptions`, but remove the `account_name` parameter from all the `Configure...` functions. That allows us to avoid the repetition of the `account_name` parameter and is more in line to how every credential constructor doesn't really need an `account_name`.

@Tom-Newton what you think and can you do this?

Reply:
That sounds good to me :+1:

Reply:
Do we actually need to make `account_name_` public though? Personally I would just make `account_name` a required argument of the AzureOptions constructor.

Reply:
This would be a good idea in normal classes, but these `*Options` classes tend to be designed like simple C structs that can be constructed from any language and easily serialized. The only exception being the secret stuff that we don't want to leak and lazily initialized fields.

Reply:
Ok. I think its as you recommended now.